src/compile/ast.rs

```diff
@@ -456,42 +618,47 @@ impl Type {
             | Type::Bytes32
             | Type::EthAddress
             | Type::Buffer
-            | Type::Every => (self == rhs),
+            | Type::Every => match rhs.rep(type_tree) {
+                Ok(right) => right == *self,
+                Err(_) => false,
+            },
             Type::Tuple(tvec) => {
-                if let Ok(Type::Tuple(tvec2)) = rhs.get_representation(type_tree) {
+                if let Ok(Type::Tuple(tvec2)) = rhs.rep(type_tree) {
                     type_vectors_assignable(tvec, &tvec2, type_tree, seen)
                 } else {
                     false
                 }
             }
             Type::Array(t) => {
-                if let Ok(Type::Array(t2)) = rhs.get_representation(type_tree) {
+                if let Ok(Type::Array(t2)) = rhs.rep(type_tree) {
                     t.assignable(&t2, type_tree, seen)
                 } else {
                     false
                 }
             }
             Type::FixedArray(t, s) => {
-                if let Ok(Type::FixedArray(t2, s2)) = rhs.get_representation(type_tree) {
+                if let Ok(Type::FixedArray(t2, s2)) = rhs.rep(type_tree) {
                     (*s == s2) && t.assignable(&t2, type_tree, seen)
                 } else {
                     false
                 }
             }
             Type::Struct(fields) => {
-                if let Ok(Type::Struct(fields2)) = rhs.get_representation(type_tree) {
+                if let Ok(Type::Struct(fields2)) = rhs.rep(type_tree) {
                     field_vectors_assignable(fields, &fields2, type_tree, seen)
                 } else {
                     false
                 }
             }
-            Type::Nominal(_, _) => {
-                if let (Ok(left), Ok(right)) = (
-                    self.get_representation(type_tree),
-                    rhs.get_representation(type_tree),
-                ) {
+            Type::Nominal(_, _, _) => {
+                if let (Ok(left), Ok(right)) = (self.rep(type_tree), rhs.rep(type_tree)) {
                     if seen.insert((left.clone(), right.clone())) {
-                        left.assignable(&right, type_tree, seen)
+                        let result = left.assignable(&right, type_tree, seen.clone());
+                        if !result {
+                            dbg!(&left, &right);
```

Comment:
This debug print should probably be removed, as it is hard to tell what it means without looking at the source code.

Reply:
Ah nice catch, accidentally left that in while debugging